### PR TITLE
Runs against aws-c-http "H1B" API

### PR DESCRIPTION
**Pull request by graebm**
This PR only updates the bindings to work against the updated HTTP API.
The external Java API has not been changed.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/native/http_request_response.c [hunk 1]
```diff
@@ -326,6 +367,7 @@ static void aws_byte_buf_transfer_best_effort(struct aws_byte_buf *dst, struct a
     src->len -= amt_to_copy;
 }
 
+// Returns AWS_OP_SUCCESS or AWS_OP_ERR
 static int aws_http_resp_body_publish_to_java(
```
Comment (justinboswell) [inline src/native/http_request_response.c]: static should be ```s_```

File: src/native/crt.c [hunk 2]
```diff
@@ -91,12 +91,14 @@ jbyteArray aws_java_byte_array_new(JNIEnv *env, size_t size) {
     return jArray;
 }
 
-void aws_copy_java_byte_array_to_native_array(JNIEnv *env, jbyteArray src, uint8_t *dst, size_t amount) {
+bool aws_copy_java_byte_array_to_native_array(JNIEnv *env, jbyteArray src, uint8_t *dst, size_t amount) {
     (*env)->GetByteArrayRegion(env, src, 0, (jsize)amount, (jbyte *)dst);
+    return (*env)->ExceptionCheck(env);
```
[the remark this answers is not in the document]
Comment (graebm) [reply src/native/crt.c]: [Docs](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#Get_PrimitiveType_ArrayRegion_routines) say:
> THROWS:
> ArrayIndexOutOfBoundsException: if one of the indexes in the region is not valid.

[the remark this answers is not in the document]
Comment (graebm) [reply src/native/crt.c]: I commented in the header file that ArrayIndexOutOfBoundsException is the only possible exception

**Comment by alexw91** [comment]
LGTM

**Pull request merged**